src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -710,13 +710,21 @@ export class SavedObjectsRepository {
     filter,
     preference,
   }: SavedObjectsFindOptions): Promise<SavedObjectsFindResponse<T>> {
-    if (!type) {
+    if (!type && !typesAndNamespacesMap) {
       throw SavedObjectsErrorHelpers.createBadRequestError(
         'options.type must be a string or an array of strings'
       );
+    } else if (type && typesAndNamespacesMap) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        'options.type must be an empty string when options.typesAndNamespacesMap is used'
+      );
```

Comment:
Should we have a similar check for `namespaces` too?

Reply:
Whoops, I should have mentioned in a comment --

I intentionally didn't implement a check for namespaces. My reasoning is that the absence of `namespaces` is treated as the default namespace, so we can't really do this check (unless we change the type signature to check for `null`, or check for an empty namespaces array...) At any rate, I think checking if `type` is falsy should accomplish what we need. What do you think?

Reply:
> I intentionally didn't implement a check for namespaces. My reasoning is that the absence of namespaces is treated as the default namespace, so we can't really do this check (unless we change the type signature to check for null, or check for an empty namespaces array...) At any rate, I think checking if type is falsy should accomplish what we need. What do you think?

Sorry for being dense -- I don't follow. Would something like this not work?

```ts
if ((namespaces?.length ?? 0) > 0 && typesAndNamespacesMap) {
  throw SavedObjectsErrorHelpers.createBadRequestError(
    'options.namespaces cannot be defined when options.typesAndNamespacesMap is used'
  );
}
```

It doesn't preclude someone from thinking they're working within the default space when `typesAndNamespacesMap` is also specified, but it at least catches the scenarios where namespaces are explicitly declared in both properties

Reply:
Er, I could have worded that better -- "so we can't really do this check" isn't true, rather I thought the check is inconsistent at that point, because an undefined namespaces field is valid otherwise.

Rather than requiring namespaces be undefined, how about we require it be an empty array? I added this in 82481d2, see what you think.

Reply:
> Rather than requiring namespaces be undefined, how about we require it be an empty array? I added this in b967f60, see what you think.

Yeah that seems reasonable to me! Thanks for clarifying your thinking